cpp/src/fil/infer.cu

```diff
@@ -309,17 +318,17 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
   int num_classes;
 
   static size_t smem_finalize_footprint(int num_classes) {
-    return sizeof(int) * num_classes * NITEMS;
```

Comment:
this is misleading, as the placement of the array was in the accumulate portion, so would have to consider the cols_shmem_size to be fully correct. But producing 0 is fine, since the lingering accumulate_footprint is accounted for when doung `max(accumulate_footprint, finalize_footprint)`